dsp/impl/magnitude_u16o12.c

```diff
@@ -0,0 +1,189 @@
+#include <stdlib.h>
+#include <math.h>
+#include <endian.h>
+#include "dsp/helpers/tables.h"
+
+/*
+ * Convert (little-endian) unsigned 16 offset 12 ( Excess 2048 format)
+ * values to unsigned 16-bit magnitudes.
+ *
+ * Offset 12:
+ *
+ *          Signed      Scaled
+ * Input    Value       Magnitude
+ * ------------------------------
+ * 0        <invalid>   <invalid>
+ * 1        -2047       65535
+ * 2        -2046       65503
+ * ...
+ * 2046     -2          64
+ * 2047     -1          32
+ * 2048      0          0
+ * 2049      1          32
+ * 2050      2          64
+ *
+ * ...
+ * 4093      2045       65471
+ * 4094      2046       65503
+ * 4095      2047       65535
+ *
+ */
+
+// The magic scaler is (32767.0 / 2047.0)
+#define MAGIC_SCALER 32.01514f
```

Comment:
Put the expression in the macro, the compiler can tell that it's constant

Reply:
done